src/Analyzers/CSharp/CodeFixes/NewLines/ConditionalExpressionPlacement/ConditionalExpressionPlacementCodeFixProvider.cs

```diff
@@ -0,0 +1,107 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.NewLines.ConditionalExpressionPlacement
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = PredefinedCodeFixProviderNames.ConditionalExpressionPlacement), Shared]
+    internal sealed class ConditionalExpressionPlacementCodeFixProvider : CodeFixProvider
+    {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public ConditionalExpressionPlacementCodeFixProvider()
+        {
+        }
+
+        public override ImmutableArray<string> FixableDiagnosticIds
+            => ImmutableArray.Create(IDEDiagnosticIds.ConditionalExpressionPlacementDiagnosticId);
+
+        public override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var document = context.Document;
+            var diagnostic = context.Diagnostics.First();
+            context.RegisterCodeFix(
+                CodeAction.Create(
+                    CSharpCodeFixesResources.Place_token_on_following_line,
+                    c => UpdateDocumentAsync(document, ImmutableArray.Create(diagnostic), c),
+                    nameof(CSharpCodeFixesResources.Place_token_on_following_line)),
+                context.Diagnostics);
+            return Task.CompletedTask;
+        }
+
+        private static async Task<Document> UpdateDocumentAsync(
+            Document document, ImmutableArray<Diagnostic> diagnostics, CancellationToken cancellationToken)
+        {
+            var text = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
+            var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            using var _ = ArrayBuilder<TextChange>.GetInstance(out var edits);
+
+            foreach (var diagnostic in diagnostics)
+            {
+                var questionToken = root.FindToken(diagnostic.Location.SourceSpan.Start);
+                Contract.ThrowIfTrue(questionToken.Kind() != SyntaxKind.QuestionToken);
```

Comment:
Consider passing a message with the actual kind we found. In case we hit this in future, it can help investigating when there is no clear repro.

Reply:
this will file a watson.  So we'll have a dump to memory examine :)

Reply:
@CyrusNajmabadi Nice to know that exceptions in codefixes will send you crash data. But how do you know if that is not a third-party codefix crashing? or you send data either way?

Reply:
we'd send data anyways.